client/utils/rest.go

```diff
@@ -101,30 +101,74 @@ func WriteGenerateStdTxResponse(w http.ResponseWriter, cdc *codec.Codec, txBldr
 // BaseReq defines a structure that can be embedded in other request structures
 // that all share common "base" fields.
 type BaseReq struct {
-	Name          string `json:"name"`
-	Password      string `json:"password"`
-	ChainID       string `json:"chain_id"`
-	AccountNumber uint64 `json:"account_number"`
-	Sequence      uint64 `json:"sequence"`
-	Gas           string `json:"gas"`
-	GasAdjustment string `json:"gas_adjustment"`
-	GenerateOnly  bool   `json:"generate_only"`
-	Simulate      bool   `json:"simulate"`
+	Name          string    `json:"name"`
+	Password      string    `json:"password"`
+	Memo          string    `json:"memo"`
+	ChainID       string    `json:"chain_id"`
+	AccountNumber uint64    `json:"account_number"`
+	Sequence      uint64    `json:"sequence"`
+	Fees          sdk.Coins `json:"fees"`
+	Gas           uint64    `json:"gas"`
```

Comment:
Wait, we actually accept "simulate" as a valid gas parameter. That triggers auto detection of gas.